### DISCUSSION TITLE
Feature: Super Layout idea (A Layout that can contain one or more of other layouts)

### DISCUSSION THREAD

**Discussion by sksar** [directus/directus#11821] [Feature Requests]
So I was thinking of a new feature that I would soon start a discussion on the GitHub discussions. 

I was thinking of a new layout which in essence would be kinda like a Super-Layout. (A Layout that hosts other layouts). Allow me to elaborate on that.

The super layout would be selectable just from the normal layout. Or it could be a core feature. Then in the same right sidebar, you can select other layouts. The sub-layouts will be in the order of selection in the layout options.
The layouts are controlled individually, and the pagination and all that is up to the layout itself. Just that the layouts would appear as sections, one after the another, and the rest stays the same.

Something like this:

![photo_2022-02-23 17 15 48](https://user-images.githubusercontent.com/11752260/155325356-bb9e88e9-2d70-4e2b-a492-e86db9a273d3.jpeg)

This is a basic example of how the super layout feature could work
The red box is a custom layout. And the table is just the core tabular layout.
The super-layout hosts these two layouts but it can host one or more layouts.

So it becomes somewhat of a modular layout structure, rather than being a single layout. The basic idea would be to see multiple layouts at the same time. Just one after the other.

So this would help to add something like the one shown in the above picture.  Currently, if we want something like that, we would basically have to recreate the entire tabular layout and then add the things we want (the red box) and that is a lot of reinventing the wheel. Add on top of that you have to keep maintaining this new layout. Which would be a lot of hassle.

Instead, if we had a system of modular super-layout structure, then we could just focus on creating just the RedBox part (in the image above), as a simple custom layout, and then using the Super-Layout feature we can use our custom layout and the inbuilt tabular layout, to achieve results like the above.

So in my opinion this kind of super layout feature opens up a lot of possibilities.

**Comment by benhaynes** [comment]
We've been discussing it for a while, but I would love to formally align on where we can add "slots" both globally and within the page/context scope. This will allow for more granular customization of existing pages without resorting to rebuilding the page as an extension (just for a small change).

**Reply by supermx1** [reply]
Perfect

**Reply by sksar** [reply]
> We've been discussing it for a while, but I would love to formally align on where we can add "slots" both globally and within the page/context scope. This will allow for more granular customization of existing pages without resorting to rebuilding the page as an extension (just for a small change).

Yes, that would also be a great idea. But as of now, it kind of made sense to me, what if someone wants multiple layouts at the same time. Wouldn't that be a common use case?

**Reply by benhaynes** [reply]
You can currently configure multiple layouts and save each as a Bookmark for immediate access. But there is no way to show two on the page at the same time (eg: a map + a calendar).

**Reply by sksar** [reply]
> You can currently configure multiple layouts and save each as a Bookmark for immediate access. But there is no way to show two on the page at the same time (eg: a map + a calendar).

Yes, two or multiple at the same time is the feature that would help.

**Reply by rijkvanzanten** [reply]
Totally, though what you're showing in your screenshot above would probably closer to inlining some insights panels rather than a secondary layout I think 🤔 (unless we'd add a new readonly style aggregated layout type)

**Reply by sksar** [reply]
> Totally, though what you're showing in your screenshot above would probably closer to inlining some insights panels rather than a secondary layout I think 🤔 (unless we'd add a new readonly style aggregated layout type)

Yes, but the attached image is just an example. It is not an exhaustive list. The use cases could be a lot, probably endless if we can do the super layout thingy. I am not looking at this feature just to solve what is shown in the picture. There are a lot more examples.

**Reply by rijkvanzanten** [reply]
Yeah I'm aware. I was thinking through a feature that would allow you to inline an insights dashboard at the top of the layout, so i was curious to see how that overlaps with this, as the ideas are very similar, but have a different use case

**Reply by sksar** [reply]
> Yeah I'm aware. I was thinking through a feature that would allow you to inline an insights dashboard at the top of the layout, so i was curious to see how that overlaps with this, as the ideas are very similar, but have a different use case

Yes, that would also be something. But how about both, like you can use both an insights module and also a layout if you shall want it. Like kinda thinking the Insights module to be another layout or vice versa. (Just a thought)
Like the red box could be whatever, an insights module or another layout. But then it should not be like only two layouts or a layout and insights module, it could be like one or more than one, in any way the user wants to configure.

**Reply by rijkvanzanten** [reply]
Yeah! I was thinking we could render a "mini workspace" with multiple configurable panels as the second half, but like you said it could technically do both. 🙂

**Reply by sksar** [reply]
> Yeah! I was thinking we could render a "mini workspace" with multiple configurable panels as the second half, but like you said it could technically do both. 🙂

Yes. That would add more modularity, rather than choosing one. More permutations and combinations will cover a greater range of use cases. The more, the merrier.

**Reply by rijkvanzanten** [reply]
> The more, the merrier.

Welllllllllllllllll don't fully agree with that. Feature bloat is a very real danger we gotta be careful about 🙂

**Reply by sksar** [reply]
> > The more, the merrier.
> 
> Welllllllllllllllll don't fully agree with that. Feature bloat is a very real danger we gotta be careful about 🙂

Yes, that is also correct. But I was talking about the possibilities a modular layout/insights structure could bring. Hence I thought about the simplest approach to just have a super layout concept, and then for additional features, people can just build their own mini custom layouts and use them in conjunction with the core/inbuilt layouts.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
